### ISSUE TITLE
require_once mbstring exception v1.10.18 on PHP 8.0

### ISSUE THREAD

**Issue by nspyke** [composer/composer#9543]
While running the Bitnami PHP-FPM 8.0 image, I get the following require_once error

```
❯ docker run --rm -it bitnami/php-fpm:8.0 bash
root@164943856e7d:/app# composer --version
Composer version 1.10.17 2020-10-30 22:31:58
```
```
root@164943856e7d:/app# composer selfupdate
A new stable major version of Composer is available (2.0.8), run "composer self-update --2" to update to it. See also https://getcomposer.org/2
Updating to version 1.10.18 (1.x channel).
   Downloading (100%)
Use composer self-update --rollback to return to version 1.10.17
```
```
root@164943856e7d:/app# composer --version
PHP Warning:  require_once(phar:///opt/bitnami/php/bin/composer.phar/vendor/symfony/polyfill-mbstring/Resources/mb_convert_variables.php8): Failed to open stream: phar error: "vendor/symfony/polyfill-mbstring/Resources/mb_convert_variables.php8" is not a file in phar "/opt/bitnami/php/bin/composer.phar" in phar:///opt/bitnami/php/bin/composer.phar/vendor/symfony/polyfill-mbstring/bootstrap.php on line 115
PHP Fatal error:  Uncaught Error: Failed opening required 'phar:///opt/bitnami/php/bin/composer.phar/vendor/symfony/polyfill-mbstring/Resources/mb_convert_variables.php8' (include_path='.:/opt/bitnami/php/lib/php') in phar:///opt/bitnami/php/bin/composer.phar/vendor/symfony/polyfill-mbstring/bootstrap.php:115
Stack trace:
#0 phar:///opt/bitnami/php/bin/composer.phar/vendor/composer/autoload_real.php(69): require()
#1 phar:///opt/bitnami/php/bin/composer.phar/vendor/composer/autoload_real.php(59): composerRequireComposerPhar1607012179()
#2 phar:///opt/bitnami/php/bin/composer.phar/vendor/autoload.php(7): ComposerAutoloaderInitComposerPhar1607012179::getLoader()
#3 phar:///opt/bitnami/php/bin/composer.phar/src/bootstrap.php(15): include('...')
#4 phar:///opt/bitnami/php/bin/composer.phar/src/bootstrap.php(18): includeIfExists()
#5 phar:///opt/bitnami/php/bin/composer.phar/bin/composer(8): require('...')
#6 /opt/bitnami/php/bin/composer.phar(24): require('...')
#7 {main}
  thrown in phar:///opt/bitnami/php/bin/composer.phar/vendor/symfony/polyfill-mbstring/bootstrap.php on line 115
```

```
root@164943856e7d:/app# composer diagnose
PHP Warning:  require_once ... etc (same error as above)
```

Tried downloading it directly
```
root@164943856e7d:/app# wget https://getcomposer.org/download/1.10.18/composer.phar
--2020-12-04 05:24:48--  https://getcomposer.org/download/1.10.18/composer.phar
Resolving getcomposer.org (getcomposer.org)... 142.44.245.229, 2607:5300:201:2100::4:d105
Connecting to getcomposer.org (getcomposer.org)|142.44.245.229|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 2000759 (1.9M) [application/octet-stream]
Saving to: 'composer.phar'

composer.phar                                                         100%[=========================================================================================================================================================================>]   1.91M   962KB/s    in 2.0s

2020-12-04 05:24:52 (962 KB/s) - 'composer.phar' saved [2000759/2000759]

root@164943856e7d:/app# php composer.phar
PHP Warning:  require_once(phar:///app/composer.phar/vendor/symfony/polyfill-mbstring/Resources/mb_convert_variables.php8): Failed to open stream: phar error: "vendor/symfony/polyfill-mbstring/Resources/mb_convert_variables.php8" is not a file in phar "/app/composer.phar" in phar:///app/composer.phar/vendor/symfony/polyfill-mbstring/bootstrap.php on line 115
PHP Fatal error:  Uncaught Error: Failed opening required 'phar:///app/composer.phar/vendor/symfony/polyfill-mbstring/Resources/mb_convert_variables.php8' (include_path='.:/opt/bitnami/php/lib/php') in phar:///app/composer.phar/vendor/symfony/polyfill-mbstring/bootstrap.php:115
Stack trace:
#0 phar:///app/composer.phar/vendor/composer/autoload_real.php(69): require()
#1 phar:///app/composer.phar/vendor/composer/autoload_real.php(59): composerRequireComposerPhar1607012179()
#2 phar:///app/composer.phar/vendor/autoload.php(7): ComposerAutoloaderInitComposerPhar1607012179::getLoader()
#3 phar:///app/composer.phar/src/bootstrap.php(15): include('...')
#4 phar:///app/composer.phar/src/bootstrap.php(18): includeIfExists()
#5 phar:///app/composer.phar/bin/composer(8): require('...')
#6 /app/composer.phar(24): require('...')
#7 {main}
  thrown in phar:///app/composer.phar/vendor/symfony/polyfill-mbstring/bootstrap.php on line 115
```

**Comment by Slamdunk** [comment]
I can confirm the bug also on Windows: https://github.com/paratestphp/paratest/runs/1497302541#step:3:16

**Comment by Seldaek** [comment]
Ah sorry forgot to backport this fix to 1.10. Fixed now in 1.10.19
